src/lib/client.ts

```diff
@@ -152,59 +47,121 @@ export class Client {
+              let blockReward = 0;
+              const allRecords: Vec<any> = await this.api.query.system.events.at(lastHeader.hash);
+              allRecords.forEach((record) => {
+                const { section, method, data } = record.event;
+                if (section === "rewards" && method === "BlockReward")
+                  blockReward = data[1] / SUNIT;
+                if (section === "transactionFees")
+                  console.log("transactionFees event::", section, method, data);
+              });
               const block: FarmedBlock = {
                 author: solution.public_key.toString(),
                 id: lastHeader.hash.toString(),
                 time: Date.now(),
                 transactions: 0,
                 blockNum: lastHeader.number.toNumber(),
-                blockReward: 0,
-                feeReward: 0,
+                blockReward,
+                feeReward: 0
               };
               this.data.farming.farmed = [block].concat(
                 this.data.farming.farmed
               );
-              storeBlocks(this.farmed);
+              this.storeBlocks(this.data.farming.farmed)
             }
           })
         process.on('beforeExit', this.do.blockSubscription.stopOnReload)
         window.addEventListener('unload', this.do.blockSubscription.stopOnReload)
         this.clearTauriDestroy = await tauri.event.once('tauri://destroyed', () => {
           console.log('Destroyed event!')
-          storeBlocks(this.data.farming.farmed)
+          this.storeBlocks(this.data.farming.farmed)
         })
       },
       stop: (): void => {
         console.log('block subscription stop triggered')
         this.unsubscribe()
         try {
           this.clearTauriDestroy()
-          storeBlocks(this.data.farming.farmed)
+          this.storeBlocks(this.data.farming.farmed)
           window.removeEventListener('unload', this.do.blockSubscription.stopOnReload)
         } catch (error) {
           console.error(error)
         }
       },
-      runTest(): void {
+      startSubcriptions(): void {
         this.start()
       }
     }
   }
   constructor() {
-    this.data.farming.farmed = this.farmed
   }
-  async init(): Promise<void> {
-    this.api = await ApiPromise.create({
-      provider: this.wsProvider, types: {
-        Solution: {
-          public_key: 'AccountId32'
-        },
-        SubPreDigest: {
-          slot: 'u64',
-          solution: 'Solution'
+
+  // If the app is started for the first time, the client will be started from PlottingProgress page.
+  // In this case, farmerPublicKey and mnemonic must be stored and clear stored farmedBlocks if exist (clearStoredBlocks).
+  // -
+  // If the app is started for the second time, the client will be started from Dashboard page.
+  // In this case, farmerPublicKey and mnemonic already exist and just need to load the stored farmedBlocks (loadStoredBlocks).
```

Comment:
When the app is started for the first time, it should start from the password setup page, then proceed into plotSetup -> plottingProgress -> Modals for extra info and private key -> Dashboard

if it is started for the second time, it should start directly from plotting progress (because probably it needs to plot some additional pieces for the initial plotting, since the farmer was offline for a bit. 
So in plotting progress, maybe it will start from 90% progress, and will plot for the remaining 10%. 
Another difference would be, we shouldn't display the info Modals for the second time.

However, I think this version is fine at the moment. We will add/update these functionalities in a later PR.

Reply:
I would think it would be better to have a plotting indicator on the main farming page, once the initial "batch" of plotting is completed. just something in the corner next to network status, could just be "plotting status" with a light, and a tooltip. 

But once again, just thoughts for a later version.

Reply:
Just as a comment, this changes does not affect the way the pages are loaded or rendered, changes only the way the client class and polkadot api is initialized and loaded.  

https://github.com/subspace/subspace-desktop/pull/39#discussion_r807440890